sql/core/src/main/scala/org/apache/spark/sql/execution/Exchange.scala

```diff
@@ -178,17 +223,14 @@ case class Exchange(
           } else {
             new ShuffledRDD[Row, Null, Null](rdd, part)
           }
-        val keySchema = child.output.map(_.dataType).toArray
-        shuffled.setSerializer(serializer(keySchema, null, numPartitions))
-
+        shuffled.setSerializer(serializer)
         shuffled.map(_._1)
 
       case SinglePartition =>
-        // SPARK-4479: Can't turn off defensive copy as what we do for `HashPartitioning`, since
```

Comment:
To be honest, I'm also confused by this comment now. Especially, `TakeOrdered.execute()` already made a defensive copy... Removing this LGTM.